### PR TITLE
Fix float division by zero when constructing ranges in AC

### PR DESCRIPTION
**Pull request by polyntsov**
`weight` equal to 1 leads to the float division by zero when constructing ranges, avoid it by adding a corresponding check. This doesn't change the observable behavior of the code.
Also, use option member `weight_` when reconstructing ranges so that the value validation check is performed for weight for which ranges are reconstructed.

### CODE REVIEW

File: src/core/algorithms/algebraic_constraints/ac_algorithm.cpp
Comment (BUYT-1) [inline src/core/algorithms/algebraic_constraints/ac_algorithm.cpp:301]: nit: I think this line should be at the beginning of the method so it fails earlier if a wrong value is passed and so that whoever reads the code sees that this sets an option.

Comment (polyntsov) [reply src/core/algorithms/algebraic_constraints/ac_algorithm.cpp:301]: Agree, good point

**Review by BUYT-1 — APPROVED** [review]
LGTM

**Comment by polyntsov** [comment]
Waiting for approve from @Sched71 before merging

**Comment by Sched71** [comment]
All good. Thank you :+1:

**Pull request merged**
